### PR TITLE
[SPARK-5013] [MLlib] Added documentation and sample data file for GaussianMixture

### PR DESCRIPTION
**Pull request by tgaloppo**
Simple description and code samples (and sample data) for GaussianMixture

### CODE REVIEW

File: docs/mllib-clustering.md [hunk 1]
```diff
@@ -168,6 +187,112 @@ print("Within Set Sum of Squared Error = " + str(WSSSE))
...
+{% highlight python %}
+from pyspark.mllib.clustering import GaussianMixture
+from numpy import array
+
+# Load and parse the data
+data = sc.textFile("data/mllib/gmm_data.txt")
+parsedData = data.map(lambda line: array([float(x) for x in line.strip().split(' ')]))
+
+# Build the model (cluster the data)
+gmm = new GaussianMixture.train(parsedData, 2)
```
Comment (mengxr) [inline docs/mllib-clustering.md]: remove `new`

File: docs/mllib-clustering.md [hunk 2]
```diff
@@ -168,6 +187,112 @@ print("Within Set Sum of Squared Error = " + str(WSSSE))
...
+            gmm.weights()[j], gmm.gaussians()[j].mu(), gmm.gaussians()[j].sigma());
+    }
+  }
+}
+{% endhighlight %}
+</div>
+
+<div data-lang="python" markdown="1">
+In the following example after loading and parsing data, we use a
+[`GaussianMixture`](api/scala/index.html#org.apache.spark.mllib.clustering.GaussianMixture)
```
Comment (mengxr) [inline docs/mllib-clustering.md]: This should be linked to the Python doc, e.g., `[NaiveBayes](api/python/pyspark.mllib.classification.NaiveBayes-class.html)`.

**Comment by mengxr** [comment]
@tgaloppo Thanks for working on the user guide! I saw this is still marked as `WIP`. Are you going to add more content?

**Comment by tgaloppo** [comment]
@mengxr I was considering adding a graphic of the test data with the recovered 2-d gaussians... I'm not sure if it would really be beneficial or not.

**Comment by tgaloppo** [comment]
@mengxr I have made the fixes you pointed out.  I am having trouble building the API docs so I can not verify that the link to the python GaussianMixture class resolves properly.

**Comment by mengxr** [comment]
Did you follow the `docs/README.md`?

**Comment by tgaloppo** [comment]
@mengxr I was able to build the docs (I had to do a clean build on my source tree for some reason).  I have checked all links (that I added, anyway).  I also updated mllib-guide.md to include Gaussian mixture (and power iteration, since it was also missing) in the list of clustering techniques.

I've removed the WIP tag; I will not add any more content, unless requested.

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-1.3! Thanks for working on the user guide!

**Pull request closed without merging**